### ISSUE TITLE
Polygon.containsPoint is incorrect at y-value of last point in a triangle

### ISSUE THREAD

**Issue by j-n-f** [ruffythepirate/ts-geometry-2d#24]
Somewhere in a project give this a try:

```ts
import { Point, Polygon } from 'ts-2d-geometry/dist';

const POINTS: Point[] = [new Point(0.153, 0.048), new Point(0.17, 0.7), new Point(0.692, 0.308)];
const TRIANGLE: Polygon = Polygon.fromPoints(POINTS);
console.debug('y = 0.306', TRIANGLE.containsPoint(new Point(0.2, 0.306)));
console.debug('y = 0.307', TRIANGLE.containsPoint(new Point(0.2, 0.307)));
console.debug('y = 0.308', TRIANGLE.containsPoint(new Point(0.2, 0.308)));
console.debug('y = 0.309', TRIANGLE.containsPoint(new Point(0.2, 0.309)));
console.debug('y = 0.310', TRIANGLE.containsPoint(new Point(0.2, 0.31)));
console.debug(
    "There's clearly something weird about the last y co-ordinate, " +
    'these are outside the left edge of the triangle'
);
console.debug('y = 0.308, x = 0.153', TRIANGLE.containsPoint(new Point(0.153, 0.308)));
console.debug('y = 0.308, x = 0.154', TRIANGLE.containsPoint(new Point(0.154, 0.308)));
console.debug('y = 0.308, x = 0.155', TRIANGLE.containsPoint(new Point(0.155, 0.308)));
console.debug('y = 0.308, x = 0.156', TRIANGLE.containsPoint(new Point(0.156, 0.308)));
console.debug('y = 0.308, x = 0.157', TRIANGLE.containsPoint(new Point(0.157, 0.308)));
console.debug('y = 0.308, x = 0.158', TRIANGLE.containsPoint(new Point(0.158, 0.308)));
```

logs:

```
y = 0.306 true
y = 0.307 true
y = 0.308 false
y = 0.309 true
y = 0.310 true
There's clearly something weird about the last y co-ordinate, these are outside the left edge of the triangle
y = 0.308, x = 0.153 true
y = 0.308, x = 0.154 true
y = 0.308, x = 0.155 true
y = 0.308, x = 0.156 true
y = 0.308, x = 0.157 true
y = 0.308, x = 0.158 true
```

Per the documentation, these points are ordered clockwise when creating the Polygon.

I thought maybe it was an issue with floating point representation of 0.308, but then realized that it's detecting points at y = 0.308 outside the left edge of the triangle as being inside the triangle. You can move that last y value up and down, and you get the same result.

**Comment by ruffythepirate** [comment]
Hey, thanks for reaching out! I'll try to take a look at this issue. I'm a bit crunched for time at the moment, but hope to be able to find some time before the end of the week

**Comment by ruffythepirate** [comment]
This issue has now been fixed in version 6.3.3. It turned out to be a bug when the point was having the same y-value as a point inside of the polygon.
